citesphere/src/main/java/edu/asu/diging/citesphere/web/user/authorities/AuthorityEntryController.java

```diff
@@ -11,31 +12,95 @@
 import edu.asu.diging.citesphere.user.IUser;
 import edu.asu.diging.citesphere.web.user.FoundAuthorities;
 
 @Controller
+@SessionScope
 public class AuthorityEntryController {
-    
+
     private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private AuthoritySearchResult authorityConceptpowerResult;
     
     @Autowired
     private IAuthorityService authorityService;
+    
+    @RequestMapping("/auth/authority/{zoteroGroupId}/find/userAuthorities")
+    public ResponseEntity<AuthoritySearchResult> getUserAuthorities(Model model, Authentication authentication,
+            @PathVariable("zoteroGroupId") String zoteroGroupId,
+            @RequestParam(defaultValue = "0", required = false, value = "page") int page,
+            @RequestParam(defaultValue = "10", required = false, value = "pageSize") int pageSize,
+            @RequestParam("firstName") String firstName, @RequestParam("lastName") String lastName) {
+
+        AuthoritySearchResult authorityResult = new AuthoritySearchResult();
+
+        List<IAuthorityEntry> userEntries = authorityService.findByName((IUser) authentication.getPrincipal(),
+                firstName, lastName, page, pageSize);
+
+        authorityResult.setFoundAuthorities(userEntries);
+        authorityResult.setCurrentPage(page + 1);
+        authorityResult.setTotalPages(authorityService
+                .getTotalUserAuthoritiesPages((IUser) authentication.getPrincipal(), firstName, lastName, pageSize));
+        return new ResponseEntity<AuthoritySearchResult>(authorityResult, HttpStatus.OK);
+    }
+    
+    @RequestMapping("/auth/authority/{zoteroGroupId}/find/searchAuthorities/{source}")
+    public ResponseEntity<AuthoritySearchResult> getAuthoritiesFromAuthorityService(Authentication authentication,
+            @PathVariable("zoteroGroupId") String zoteroGroupId, @PathVariable("source") String source,
+            @RequestParam(defaultValue = "0", required = false, value = "page") int page,
+            @RequestParam(defaultValue = "20", required = false, value = "pageSize") int pageSize,
+            @RequestParam("firstName") String firstName, @RequestParam("lastName") String lastName) {
+
+        if ((firstName == null || firstName.isEmpty()) && (lastName == null || lastName.isEmpty())) {
+            logger.warn(
+                    "At least one of the fields must be non-empty. firstName and lastName are empty " + zoteroGroupId);
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+        AuthoritySearchResult searchResult = null;
+        try {
+            searchResult = authorityService.searchAuthorityEntries((IUser) authentication.getPrincipal(), firstName,
+                    lastName, source, page, pageSize);
+            searchResult.setCurrentPage(page + 1);
+            if(source.equalsIgnoreCase("conceptpower")) {
+                authorityConceptpowerResult = searchResult;
+            }
```

Comment:
this method should work for any number of authority services, you don't want to adjust it every time we add another one. I would use a dictionary that maps the source to the results and use that (no if statement to check what the source is).